ui/src/app/store/node/types.ts

```diff
@@ -0,0 +1,82 @@
+import type { Model } from "app/store/types";
+
+/**
+ * A named foreign model reference, e.g. machine.domain
+ */
+export type ModelRef = Model & {
+  name: string;
+};
+
+type TestResult = -1 | 0 | 1;
+
+export type TestStatus = {
+  status: TestResult;
+  pending: TestResult;
+  running: TestResult;
+  passed: TestResult;
+  failed: TestResult;
+};
+
+/**
+ * SimpleNode represents the intersection of Devices, Machines and Controllers
+ */
+export type SimpleNode = Model & {
+  actions: string[];
+  domain: ModelRef;
+  hostname: string;
+  fqdn: string;
+  link_type: string;
+  node_type_display: string;
+  permissions: string[];
+  system_id: string;
+  tags: string[];
+};
+
+type NodeStatus =
+  | "New"
+  | "Commissioning"
+  | "Failed commissioning"
+  | "Missing"
+  | "Ready"
+  | "Reserved"
+  | "Allocated"
+  | "Deploying"
+  | "Deployed"
+  | "Retired"
+  | "Broken"
+  | "Failed deployment"
+  | "Releasing"
+  | "Releasing failed"
+  | "Disk erasing"
+  | "Failed disk erasing"
+  | "Rescue mode"
+  | "Entering rescue mode"
+  | "Failed to enter rescue mode"
+  | "Exiting rescue mode"
+  | "Failed to exit rescue mode"
+  | "Testing"
+  | "Failed testing";
+
+/**
+ * Node represents the intersection of Machines and Controllers
+ */
+export type BaseNode = SimpleNode & {
```

Comment:
This was previously named `Node` but that conflicts with the DOM Node feature.

Reply:
Ah, out of interest, how did you realise that? I wasn't seeing any warnings or eslint errors. The docstring does need updating here to reflect the new name.

Reply:
When I moved the types around it didn't complain about missing variables and the TypeScript hints were for the DOM Node type.